hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1557,6 +1557,10 @@ public boolean getHbaseIndexUpdatePartitionPath() {
     return getBooleanOrDefault(HoodieHBaseIndexConfig.UPDATE_PARTITION_PATH_ENABLE);
   }
 
+  public int getHBaseIndexRegionCount() {
+    return getInt(HoodieHBaseIndexConfig.BUCKET_NUMBER);
+  }
+
```

Comment:
I have a glance at this PR. IMO, the HBase config option should not intrude into the common write config. WDYT?

Reply:
I just follow the old way in other HoodieIndex, seems they all have several getter methods in HoodieWriteConfig and only pass HoodieWriteConfig in the constructor. Do you think we need to refactor them?

Reply:
Yes, I feel like this class is a hodgepodge and is ballooning. It does not demonstrate the benefits of configuration separation that a pluggable architecture should bring. WDYT about it? @xushiyan